projects/core/directives/dropdown/dropdown.component.ts

```diff
@@ -64,9 +77,11 @@ export class TuiDropdownComponent implements OnInit {
         .closest('[tuiTheme]')
         ?.getAttribute('tuiTheme');
 
+    protected readonly styles$ = new BehaviorSubject<Partial<DynamicStyles>>({});
```

Comment:
There's no need to introduce another subject and subscription, just instead of `subscribe` with `update` in original subscription you can move `isConnected` check to `tap` (or `takeWhile` + close on complete) and `update` can be changed into `map` so it stays in the same stream and `Object.assign` would then be performed in `next` of it. I also wonder if `ngOnInit` can be changed to `startWith` inside `TuiPositionService` so it immediately reports size upon subscription instead of waiting for the next animation frame.

Reply:
Updated it accordingly.
Tried to add `startWith()` into TuiPositionService, but it makes hint behave weird, so kept `ngOnInit()` logic.